store/tikv/lock_resolver.go

```diff
@@ -342,7 +343,9 @@ func (lr *LockResolver) resolveLocks(bo *Backoffer, callerStartTS uint64, locks
 				cleanTxns[l.TxnID] = cleanRegions
 			}
 
-			if l.LockType == kvrpcpb.Op_PessimisticLock {
+			if config.GetGlobalConfig().TiKVClient.EnableAsyncCommit && status.secondaries != nil && len(status.secondaries) > 0 {
```

Comment:
I think we need to change the `getTxnStatusFromLock` function(the crash recovery logic including querying all the secondaries) which is the implementation of `checkTxnStatus` API. So that an async commit transaction status is certain after `getTxnStatusFromLock` is called, the resolve logic could work as before that it just resolves locks met.

Reply:
The transaction commit logic in 2pc may have some similar logic with crash recovery in `getTxnStatusFromLock ` or `checkTxnStatus `, maybe we could do it together.

Reply:
So, iiuc, with existing transactions we only resolve locks lazily, that is if we find an expired lock TiDB will resolve that lock, but it won't find every other lock in that transaction (if we don't use the 'lite' mechanism, TiKV will find locks in the same region, but other locks won't be resolved). With async commit, we have to do a lot of work to recover the transaction status, so in this case I think it is worth eagerly resolving all the locks. That means that the logic for calling `resolveLocks` is not the same as in the non-async case and so we can't just change `getTxnStatusFromLock`.

Does it make sense, do you think?

Reply:
For such a scenario like a large transaction holding many locks(the maximum transaction size is 10GB now) and needs recovery, other point gets blocked by this large transaction will try to resolve all the locks belonging to it so that it could read, there may be a lot of batch get requests and duplicate resolve requests. So maybe we should resolve keys blocking current transactions?

Reply:
I think that we will limit async commit to a small number of locks at first, e.g., 8 or 16 (or even 2, to be very conservative). We'll implement that check soon. Longer term we should investigate the trade-off more deeply. We can also implement some deduplication in TiKV.